vhdbuilder/scripts/automate_version_bump.sh

```diff
@@ -47,10 +49,47 @@ update_image_version() {
     sed -i "s/${current_image_version}/${new_image_version}/g" pkg/agent/datamodel/sig_config_test.go
 }
 
-find_current_image_version "pkg/agent/datamodel/osimageconfig.go"
-set_git_config
-create_branch $branch_name
-update_image_version
+create_image_bump_pr() {
+    create_branch $branch_name
+    update_image_version
 
-set +x
-create_pull_request $new_image_version $github_access_token $branch_name $pr_title
\ No newline at end of file
+    set +x
+    create_pull_request $new_image_version $github_access_token $branch_name $pr_title
+    set -x
+}
+
+# This function cuts the official branch based off the commit ID that the builds were triggered from and tags it
+cut_official_branch() {
+    official_branch_name="official/v${new_image_version//.}"
+    official_tag="v0.${new_image_version//.}.0"
+    final_commit_hash=""
+    for build_id in $build_ids; do
+        current_build_commit_hash=$(az pipelines runs show --id $build_id | jq -r '.sourceVersion')
+        if [[ -z "$final_commit_hash" ]]; then
+            final_commit_hash=$current_build_commit_hash
+        else
+            if [[ $final_commit_hash != $current_build_commit_hash ]]; then
+                echo "Builds are not triggered off the same commit, exit"
+                exit 1
+            fi
+        fi
+    done
+    echo "All builds are based off the same commit"
+
+    # Checkout branch and commit the image bump file diff to official branch too
+    git checkout -b $official_branch_name $commit_hash
+    update_image_version
+    git add .
+    git commit -m"Update image version in official branch"
+    git push -u origin $official_branch_name
+
+    git tag $official_tag
+    git push origin tag $official_tag
+    git checkout master
+
+}
+
+set_git_config
+find_current_image_version "pkg/agent/datamodel/osimageconfig.go"
```

Comment:
should we consider using sig_config here rather than osimageconfig? I see the update func hits all files, but since AB is only source of truth for sigconfig, maybe better to rely on that?

Reply:
would the 1604 freeze also affect the automated PR bump?  This PR https://github.com/Azure/AgentBaker/pull/1326/files sets  `sig_config.go` to have a particular image version for 1604 and `update_image_version` replaces the versions in the same file. It may not affect this PR but I just wanted to flag it to be sure.

Reply:
@alexeldeib is it ok if I iterate on this is a separate PR? it would require me to change the logic of extracting the current image version which depends on the file composition.

Reply:
@ganeshkumar5699 it will be a one time thing, as soon as the version for 1604 and the latest version for other SKUs are out of sync, shouldnt matter.

Reply:
Sounds good. I see how manually fixing the image version in the next VHD release would make the 1604 image correct for that and subsequent releases (as the `current_image_version` would change afterwards).